core/src/main/java/com/oath/oak/EntrySet.java

```diff
@@ -0,0 +1,951 @@
+        int valuePosition = getPositionFromIntArray(valueArray);
+        int length = getValueLengthFromIntArray(valueArray);
+
+        valueRef.setReference(blockID, valuePosition, length);
+        return true;
+    }
+
+    /**
+     * releaseKey releases key in slice, currently in use only for unreached keys,
+     * waiting for GC to be arranged
+     **/
+    void releaseKey(int ei) {
+        long keyReference = getKeyReference(ei);
+        int[] keyArray = UnsafeUtils.longToInts(keyReference);
+        int blockID = getKeyBlockIDFromIntArray(keyArray);
+        int keyPosition = getPositionFromIntArray(keyArray);
+        int length = getKeyLengthFromIntArray(keyArray);
+
+        Slice s = new Slice(blockID, keyPosition, length, memoryManager);
+        memoryManager.releaseSlice(s);
+    }
+
+    /**
+     * releaseValue releases value in slice, currently the method is used only to release an
+     * unreachable value reference, the one that was not yet attached to an entry!
+     * The method is part of EntrySet, because it cares also
+     * for writing the value before attaching it to an entry (writeValueStart/writeValueCommit)
+     **/
+    void releaseValue(OpData opData) {
+        memoryManager.releaseSlice(
+            buildValueSlice(opData.newValueReference, INVALID_VERSION, memoryManager));
+    }
+
+    /**
+     * writeValueStart writes value off-heap. Supposed to be for entry index "ei",
+     * but this entry metadata is not updated in this method. This is an intermediate step in
+     * the process of inserting key-value pair, it will be finished with writeValueCommit.
+     * The off-heap header is initialized in this function as well.
+     *
+     * @param lookUp the structure that follows the operation since the key being found.
+     *               Needed here for the entry index (of the entry to be updated
+     *               when insertion of this key-value pair will be committed) and for the old version.
+     * @param value the value to write off-heap
+     * @param writeForMove
+     * @return OpData to be used later in the writeValueCommit
+     **/
+    OpData writeValueStart(LookUp lookUp, V value, boolean writeForMove) {
+        // the length of the given value plus its header
+        int valueLength = valueSerializer.calculateSize(value) + valOffHeapOperator.getHeaderSize();
+
+        // The allocated slice is actually the thread's ByteBuffer moved to point to the newly
+        // allocated slice. Version in time of allocation is set as part of the slice data.
+        Slice slice = memoryManager.allocateSlice(valueLength, MemoryManager.Allocate.VALUE);
+
+        // for value written for the first time:
+        // initializing the off-heap header (version and the lock to be free)
+        // for value being moved, initialize the lock to be locked
+        if (writeForMove) { valOffHeapOperator.initLockedHeader(slice);}
+        else { valOffHeapOperator.initHeader(slice); }
```

Comment:
you can choose some code style format, add it to the IDE and it will help you maintain the code consistent. 
The same format should be used by all committers otherwise you will get big PR anytime someone makes even the smallest change in the document.

Reply:
another PR for code styling